### PR TITLE
xw_plus_b mkldnn implementation

### PR DESCRIPTION
**Pull request by olegsemeniv**
**libnd4j implementation of xw_plus_b op on mkldnn and xw_plus_b_bp**

Tested:
unit tests added

### CODE REVIEW

**Review by shyrma — DISMISSED** [review]

**Review by shyrma — COMMENTED** [review]
```
    if (bShouldTrans) {
                dLdw->permutei({ 1,0 });
            }
            matmul_bp mmul_bp;
            mmul_bp.execute({ x, w, dLdz }, std::vector<NDArray*>{dLdx, dLdw}, {}, {}, {});

            if (bShouldTrans) {
                dLdw->permutei({ 1,0 });
            }
```
please do not perform manipulations on input arrays which may force buffer change

File: libnd4j/tests_cpu/layers_tests/DeclarableOpsTests5.cpp [hunk 1]
```diff
@@ -2432,29 +2432,145 @@ TEST_F(DeclarableOpsTests5, ZeroFraction_3) {
...
+}
+////////////////////////////////////////////////////////////////////////////////
+TEST_F(DeclarableOpsTests5, XWPlusB_6) {
+
+    auto x = NDArrayFactory::create<float>('c', { 3, 2 }, { 1.f, 11.f,  3.f, 14.f,  5.f,  6.f });
+    auto y = NDArrayFactory::create<float>('c', { 2, 1 }, { 11.f,  3.f });
+
+    auto b = NDArrayFactory::create<float>('c', { 1 }, { 100.f });
+
+    auto exp = NDArrayFactory::create<float>('c', { 3, 1 }, { 144.000000, 175.000000, 173.000000 });
```
Comment (raver119) [inline libnd4j/tests_cpu/layers_tests/DeclarableOpsTests5.cpp]: .f literal is missing here

File: libnd4j/tests_cpu/layers_tests/DeclarableOpsTests5.cpp [hunk 2]
```diff
@@ -2432,29 +2432,145 @@ TEST_F(DeclarableOpsTests5, ZeroFraction_3) {
...
 
-    auto x = NDArrayFactory::create<double>('c', {2,3}, { 1.f, 11.f,  3.f, 14.f,  5.f,  6.f});
-    auto y = NDArrayFactory::create<double>('c', {3,2}, { 11.f,  3.f, 4.f,  5.f, 6.f,  2.f});
-    auto b = NDArrayFactory::create<double>({100.f, 200.f});
+    auto x = NDArrayFactory::create<float>('c', { 2,3 }, { 1.f, 11.f,  3.f, 14.f,  5.f,  6.f });
+    auto y = NDArrayFactory::create<float>('c', { 3,2 }, { 11.f,  3.f, 4.f,  5.f, 6.f,  2.f });
+    auto b = NDArrayFactory::create<float>({ 100.f, 200.f });
 
-    auto exp = NDArrayFactory::create<double>('c', {2,2}, {173.f, 264.f, 310.f, 279.f});
+    auto exp = NDArrayFactory::create<float>('c', { 2,2 }, { 173.f, 264.f, 310.f, 279.f });
```
Comment (raver119) [inline libnd4j/tests_cpu/layers_tests/DeclarableOpsTests5.cpp:2442]: .f literal is missing here.

File: libnd4j/tests_cpu/layers_tests/DeclarableOpsTests5.cpp [hunk 3]
```diff
@@ -2432,29 +2432,145 @@ TEST_F(DeclarableOpsTests5, ZeroFraction_3) {
...
 
     sd::ops::xw_plus_b op;
-    auto result = op.evaluate({&x, &y, &b}, {}, {});
+    auto result = op.evaluate({ &x, &y, &b }, {}, {});
```
Comment (raver119) [inline libnd4j/tests_cpu/layers_tests/DeclarableOpsTests5.cpp]: you don't need empty parenthesis if you don't have arguments

**Review by raver119 — CHANGES_REQUESTED** [review]
LGTM, few minor fixes required

File: libnd4j/tests_cpu/layers_tests/DeclarableOpsTests5.cpp (hunk 1 above)
Comment (olegsemeniv) [reply libnd4j/tests_cpu/layers_tests/DeclarableOpsTests5.cpp]: apply

File: libnd4j/tests_cpu/layers_tests/DeclarableOpsTests5.cpp (hunk 3 above)
Comment (olegsemeniv) [reply libnd4j/tests_cpu/layers_tests/DeclarableOpsTests5.cpp]: thanks, will take into account, changed

**Pull request merged**